dex/networks/dcr/script.go

```diff
@@ -701,85 +508,64 @@ func DataPrefixSize(data []byte) uint8 {
 	return uint8(prefixSize)
 }
 
-// Size is the serialized size of the input.
-func (nfo *SpendInfo) Size() uint32 {
-	return TxInOverhead + uint32(wire.VarIntSerializeSize(uint64(nfo.SigScriptSize))) + nfo.SigScriptSize
-}
-
 // InputInfo is some basic information about the input required to spend an
-// output. The pubkey script of the output is provided. If the pubkey script
-// parses as P2SH or P2WSH, the redeem script must be provided.
+// output. Only P2PKH and P2SH pkScripts are supported. If the pubkey script
+// parses as P2SH, the redeem script must be provided.
 func InputInfo(version uint16, pkScript, redeemScript []byte, chainParams *chaincfg.Params) (*SpendInfo, error) {
-	scriptType := ParseScriptType(version, pkScript, redeemScript)
-	if scriptType == ScriptUnsupported {
+	scriptType, scriptAddrs := ExtractScriptAddrs(version, pkScript, chainParams)
+	switch {
+	case scriptType == ScriptUnsupported:
 		return nil, dex.UnsupportedScriptError
+	case scriptType.IsP2PKH():
+		return &SpendInfo{
+			SigScriptSize: P2PKHSigScriptSize,
+			ScriptAddrs:   scriptAddrs,
+			ScriptType:    scriptType,
+		}, nil
```

Comment:
I didn't *need* to rework `InputInfo` this much, but nothing of substance is changed and it is far more readable IMO.